mkt/users/models.py

```diff
@@ -147,7 +137,7 @@ def is_developer(self):
 
     @property
     def name(self):
-        return smart_unicode(self.display_name or self.username)
+        return smart_unicode(self.display_name or '-')
```

Comment:
I do wonder if we should be using a default of the name part of name@domain in the email, or a pseudo random string.  (Or even something like 'user-%d' % self.id).  Otherwise the places where this is used might start to look a little odd if there are many users all with the same display name '-'.